server/graphql/v2/query/collection/ExpensesCollectionQuery.ts

```diff
@@ -372,6 +372,18 @@ const ExpensesCollectionQuery = {
     return {
       nodes: result.rows,
       totalCount: result.count,
+      totalAmount: async () => {
+        const totalExpenses = await models.Expense.findOne({
+          attributes: [[Sequelize.fn('SUM', Sequelize.col('amount')), 'amount']],
+          include,
+          where,
+          raw: true,
+        });
+
+        return {
+          value: totalExpenses.amount,
+        };
+      },
```

Comment:
1. We need to handle expenses that are not in the currency of the collective
2. We must make sure we actually return a `currency` for the amount object
3. Rather than trigger a second query, could we reuse the rows that have already been fetched

```suggestion
      async totalAmount: () => { 
        const expensesByCurrency = groupBy(result.rows, 'currency');
        const amountByCurrency = mapValues(expensesByCurrency, expenses => sumBy(expenses, 'amount'));
        // TODO: Convert all amounts to collective's currency, ideally in a single async call using req.loaders.CurrencyExchangeRate.convert.loadMany
        return { value, currency: collective.currency };
      }
```

Reply:
We can't reuse the first query as it does not contain the entire set of expenses, it contains at most a page, and sometimes no records at all when the client sets `limit: 0` to get totals.